clang/lib/CIR/CodeGen/CIRGenItaniumCXXABI.cpp

```diff
@@ -2288,6 +2288,23 @@ static mlir::cir::FuncOp getItaniumDynamicCastFn(CIRGenFunction &CGF) {
   return CGF.CGM.createRuntimeFunction(FTy, "__dynamic_cast");
 }
 
+mlir::cir::DynamicCastToVoidInfoAttr
+CIRGenItaniumCXXABI::buildDynamicCastToVoidInfo(CIRGenFunction &CGF,
+                                                QualType SrcRecordTy) {
+  mlir::Type vtableElemTy;
+  uint64_t vtableElemAlign;
+  if (CGM.getItaniumVTableContext().isRelativeLayout()) {
+    vtableElemTy = CGF.getBuilder().getSInt32Ty();
+    vtableElemAlign = 4;
+  } else {
+    vtableElemTy = CGF.ConvertType(CGF.getContext().getPointerDiffType());
+    vtableElemAlign = CGF.getPointerAlign().getQuantity();
+  }
+
+  return mlir::cir::DynamicCastToVoidInfoAttr::get(vtableElemTy,
```

Comment:
If you'd still need a way to differentiate casts to void, you could perhaps add a new `DCK_PtrCastToVoid` kind, or something.

Reply:
Updated.